### ISSUE TITLE
newbie question

### ISSUE THREAD

**Issue by wilbertttt** [mono/CppSharp#1082]
##### Brief Description

I coded a c++ .h file and .cpp file using VS 2015. Try to convert to C#.

Throw an error: 
C:\Windows\system32>mono "C:\1\CPPSHARP\CPPSHARP\bin\Debug\CPPSHARPNP.exe"
* Assertion: should not be reached at ..\mono\metadata\metadata.c:2846

OS: Windows / OS X / Linux (include version and/or distro)

##### Used headers

struct SpriteType
{
	SpriteType();
	void SetValue(float, float, float, float);
	int IsShotBy(SpriteType*);
private:
	float posX, posY, shotRadius, sensedRadius;
	float _temp0;
};

cpp file:

#include "stdafx.h"
#include<cmath>
#include"stg.h"

SpriteType::SpriteType() :posX(0.0f), posY(0.0f), shotRadius(0.0f), sensedRadius(0.0f)
{

}

void SpriteType::SetValue(float x, float y, float rshot, float rsensor)
{
	posX = x, posY = y, shotRadius = rshot, sensedRadius = rsensor;
}

int SpriteType::IsShotBy(SpriteType *pOtherSprite)
{
	_temp0 = sqrtf((pOtherSprite->posX - posX)*(pOtherSprite->posX - posX) + (pOtherSprite->posY - posY)*(pOtherSprite->posY - posY));
	return (int)(_temp0 < shotRadius);
}

##### Used settings
public void Setup(Driver driver)
        {
            var options = driver.Options;
            options.GeneratorKind = GeneratorKind.CSharp;
            options.OutputDir = @"c:\Sample";

            var module = options.AddModule("Test");
            module.IncludeDirs.Add(@"C:\Desktop\Test\Test");
            module.Headers.Add("stg.h");
        }

Target: MSVC/GCC/Clang

Other settings

**Comment by ddobrev** [comment]
C++# is not tested at all with Mono for Windows. However, it works just fine with Microsoft .NET so I would encourage you to try that instead.

**Comment by wilbertttt** [comment]
@ddobrev, thanks for your reply. My requirement is that I need invoke C++ methods in C# project.
So I need to convert C++ to C# Managed class and methods. CppSharp is used to do this, right? 
I have learned GettingStarted.md and GeneratingBindings.md, but these steps really confuse me.
I have .h, .cpp, .lib files to test. I coded a C# convert project using VS 2015 as attached, and  implement Setup(Driver driver) method. Your meaning is that running this directly, not with Mono? Output .cs file?
![vs](https://user-images.githubusercontent.com/38850747/39502960-b9b42580-4df5-11e8-8cfa-51685b668b81.png)
